src/pudl/workspace/setup.py

```diff
@@ -4,310 +4,120 @@
-    pudl_settings["ferc2_xbrl_datapackage"] = pathlib.Path(
-        pudl_settings["pudl_out"], "ferc2_xbrl_datapackage.json"
-    )
-    pudl_settings["ferc2_xbrl_taxonomy_metadata"] = pathlib.Path(
-        pudl_settings["pudl_out"], "ferc2_xbrl_taxonomy_metadata.json"
-    )
-
-    ferc6_db_file = pathlib.Path(pudl_settings["pudl_out"], "ferc6_xbrl.sqlite")
-    pudl_settings["ferc6_xbrl_db"] = "sqlite:///" + str(ferc6_db_file.resolve())
-    pudl_settings["ferc6_xbrl_datapackage"] = pathlib.Path(
-        pudl_settings["pudl_out"], "ferc6_xbrl_datapackage.json"
-    )
-    pudl_settings["ferc6_xbrl_taxonomy_metadata"] = pathlib.Path(
-        pudl_settings["pudl_out"], "ferc6_xbrl_taxonomy_metadata.json"
-    )
-
-    ferc60_db_file = pathlib.Path(pudl_settings["pudl_out"], "ferc60_xbrl.sqlite")
-    pudl_settings["ferc60_xbrl_db"] = "sqlite:///" + str(ferc60_db_file.resolve())
-    pudl_settings["ferc60_xbrl_datapackage"] = pathlib.Path(
-        pudl_settings["pudl_out"], "ferc60_xbrl_datapackage.json"
-    )
-    pudl_settings["ferc60_xbrl_taxonomy_metadata"] = pathlib.Path(
-        pudl_settings["pudl_out"], "ferc60_xbrl_taxonomy_metadata.json"
-    )
-
-    ferc714_db_file = pathlib.Path(pudl_settings["pudl_out"], "ferc714_xbrl.sqlite")
-    pudl_settings["ferc714_xbrl_db"] = "sqlite:///" + str(ferc714_db_file.resolve())
-    pudl_settings["ferc714_xbrl_datapackage"] = pathlib.Path(
-        pudl_settings["pudl_out"], "ferc714_xbrl_datapackage.json"
-    )
-    pudl_settings["ferc714_xbrl_taxonomy_metadata"] = pathlib.Path(
-        pudl_settings["pudl_out"], "ferc714_xbrl_taxonomy_metadata.json"
-    )
-
-    pudl_settings["pudl_db"] = "sqlite:///" + str(
-        pathlib.Path(pudl_settings["pudl_out"], "pudl.sqlite")
-    )
-
-    pudl_settings["censusdp1tract_db"] = "sqlite:///" + str(
-        pathlib.Path(pudl_settings["pudl_out"], "censusdp1tract.sqlite")
-    )
-    return pudl_settings
-
-
-def init(pudl_settings: dict[str, str], clobber=False):
+    @property
+    def settings_dir(self) -> Path:
+        """Path to directory containing settings files."""
+        return self.input_dir.parent / "settings"
+
+    @property
+    def data_dir(self) -> Path:
+        """Path to PUDL data directory."""
+        # TODO(janrous): possibly deprecate this in favor of input_dir
+        return self.input_dir
+
+    @property
+    def pudl_db(self) -> Path:
+        """Returns url of locally stored pudl sqlite database."""
+        return self.sqlite_db("pudl")
+
+    def sqlite_db(self, name: str) -> str:
+        """Returns url of locally stored pudl slqlite database with given name.
+
+        The name is expected to be the name of the database without the .sqlite
+        suffix. E.g. pudl, ferc1 and so on.
+        """
+        db_path = PudlPaths().output_dir / f"{name}.sqlite"
+        return f"sqlite:///{db_path}"
+        return self.output_dir / f"{name}.sqlite"
```

Comment:
Right, I wanted to have this conversation. I think that dumping all the dataset-specific filenames in this global config is a bit of an anti-pattern. It breaks the isolation in that every time you add new dataset, the associated metadata (incl. filenames) will be encoded in the dataset specific objects (e.g. configs, dataset descriptors and so on) and then you will have some of the metadata dumped in this grab-bag config here.

It would be IMO preferable if we could move the generation of filenames into dataset-specific objects/metadata. This object/function could access `PudlPaths` to construct absolute pathnames. I'm not quite sure where this would naturally fit so I haven't done it yet, but this would be the ideal direction to take this in.

WDYT?

Reply:
For example (unsure if `DatasetSettings` is the right place to put this), we could consider something along the lines of:

```python
class Ferc1Settings(GenericDatasetSettings):
  ...
  def get_output_database(self) -> Path:
    return PudlPaths().output_dir() / "ferc1.sqlite"
```

Reply:
@jdangerx do you have thoughts here?  I can see generating these paths based on metadata that's stored within the dataset being a good idea, but I'm nervous that if we don't set the paths up to be correct now (even if they're hard coded inside the PudlPaths for now) we may not get around to fixing the paths later.

Also given that we have the JSON version of the XBRL metadata, the SQLite DB outputs, the datapackage.json files, and in some cases the Parquet outputs, it seems like a substantial amount of information to potentially add to the dataset metadata -- and it's information about where the data ends up, rather than where it comes from, which is a bit different.

If we specify the paths or rules for generating them inside `PudlPaths` now but access them via a method, we can change where that information comes from later if/when we add it to the dataset metadata.

Reply:
The problem imo is that PudlPath becomes an object that needs to know a little bit about every dataset and that goes against modularity in a big way. The output paths are output_dir + base_filename. PudlPaths should be responsible for the first part but it should be dataset specific method/object that generates the latter (and it could easily combine its stuff w/ PudlPaths(). output_dir). 

I'm happy to do this refactor as part of this PR, we just have to decide on where to put it.

In theory, we could even make it Ferc1ToSqliteSettings.output_db. This may not be entirely clean but perhaps acceptable? Settings should be available where you need to connect to the output files and it could reach into PudlPaths for directory where this lives.

Reply:
Jan's vision of 
> PudlPaths should be responsible for the first part but it should be dataset specific method/object that generates the latter (and it could easily combine its stuff w/ PudlPaths(). output_dir).
makes a lot of sense to me - I know that when working with the previous settings object, generating all these paths in one place got pretty confusing.

Zane, is your point here:
> Also given that we have the JSON version of the XBRL metadata, the SQLite DB outputs, the datapackage.json files, and in some cases the Parquet outputs, it seems like a substantial amount of information to potentially add to the dataset metadata -- and it's information about where the data ends up, rather than where it comes from, which is a bit different.

that basically, each DatasetSettings would end up with a different set of output paths? Like the `Ferc1Settings` would have a `.dbf_db`, a `.xbrl_db`, a `.datapackage`, and a `.taxonomy_metadata`... and then `CensusDP1Settings` would just have a `.sqlite_db`? That seems fine, since each different dataset settings is a different class.

My vote would be for each subclass of `GenericDatasetSettings` to then include the dataset-specific output paths, and each different dataset has its own set of output paths which makes sense.

I understand the worry about not following up on the refactor, but I think this PR has lingered long enough that we should probably merge this and then do a follow-up.

Reply:
I can also see that the configs are easier to _access_ when they're all in one place, but easier to _generate_ when they're set up separately. What do you think of some sort of container object that holds *all* the configs @rousik @zaneselvans ?

Reply:
Can you clarify what you mean by all the configs? Are we talking about the etl settinings, PudlPaths and other such configurations?

As it stands now, I'm expecting that `PudlPaths()` can simply be instantiated wherever it is needed (it reads env variables and so it should always give you the right answer) so there's no need to pass that through.

AFAIK there's really one root etl settings from which you can pull out sub-components (e.g. `FercToSqliteSettings`) so there is an enclosing configuration.

There are few stray configuration fields that are standalone (e.g. datastore configuration fields such as `gcs_cache_path` and `sandbox`; or `clobber` field that seems to be set in various parts of the system).

It would be good idea to consolidate all of these into some enclosing structures, possibly even embed those in the etl settings object, but this can be done independently of this PR.

Let me know if I understand you point here or whether you were thinking about something else here.

Reply:
I still think that the best approach might be having per-dataset configuration object (akin to `PudlPaths`) that would implicitly construct absolute paths by combining `PudlPaths` base paths with the dataset specific relative paths/filenames. It might make sense to have few global paths embedded in `PudlPaths` (e.g. pudl database).

I think that single object that aggregates everything from across all datasets is going to be tricky to scale with number of supported datasets. That said, it might make sense to use _registry pattern_ here and have a way to retrieve dataset specific config via dataset name.

Reply:
The above approach maintains modularity, but allows central access to all of these via the registry in situations where appropriate.

Reply:
Yeah, a registry would be a nice way to not have to keep aggregating settings into some central SuperConfig object!

What do you think of having the `DatasetSettings` construct the dataset paths, vs. having a new `DatasetPaths` object that just handles those paths?